base/div.jl

```diff
@@ -133,62 +132,76 @@ julia> divrem(7,3)
 ```
 """
 divrem(x, y) = divrem(x, y, RoundToZero)
+
+function divrem(a::T, b::T, r::RoundingMode) where {T<:AbstractFloat}
+    if r === RoundToZero
+        (div(a,b), rem(a,b))
+    elseif r === RoundDown
+        (fld(a, b),mod(a,b))
+    else
+        (div(a,b,r), rem(a,b,r))
+    end
+end
+#avoids calling rem for Integers, Rationals
 function divrem(a, b, r::RoundingMode)
     if r === RoundToZero
         # For compat. Remove in 2.0.
-        (div(a, b), rem(a, b))
+        d = div(a, b)
+        (d, a - d*b)
     elseif r === RoundDown
         # For compat. Remove in 2.0.
-        (fld(a, b), mod(a, b))
+        d = fld(a, b)
+        (d, a - d*b)
     else
-        (div(a, b, r), rem(a, b, r))
+        d = div(a, b, r)
+        (d, a - b*d)
     end
 end
 function divrem(x::Integer, y::Integer, rnd::typeof(RoundNearest))
     (q, r) = divrem(x, y)
     if x >= 0
         if y >= 0
-            r >=        (y÷2) + (isodd(y) | iseven(q)) ? (q+true, r-y) : (q, r)
+            r >=  (y÷2) + (isodd(y) | iseven(q)) ? (q+true, r-y) : (q, r)
```

Comment:
As mentioned, these are unrelated style changes. The last person who wrote this  intended to line up `(y÷2)`. Since you're not modifying the code in these methods,  it's best that  these whitespace changes  be reverted.